src/basic/os-util.c

```diff
@@ -170,13 +170,19 @@ int open_extension_release(const char *root, const char *extension, char **ret_p
                         }
                 }
         } else {
-                FOREACH_STRING(p, "/etc/os-release", "/usr/lib/os-release") {
-                        r = chase_symlinks(p, root, CHASE_PREFIX_ROOT,
+                const char *var = secure_getenv("SYSTEMD_OS_RELEASE");
+                if (var)
```

Comment:
i wonder if it wouldn't be better to ignore the env var if root is set?

Reply:
The opposite: I think it's particularly useful when operating with `--root`, because you might have os-release that is not set up yet, or is only partially filled in. It's also quite useful for testing.